src/compiler/binder.ts

```diff
@@ -2163,13 +2164,16 @@ namespace ts {
                 case SyntaxKind.ModuleBlock:
                     return updateStrictModeStatementList((<Block | ModuleBlock>node).statements);
 
+                case SyntaxKind.JSDocParameterTag:
+                    if (node.parent.kind !== SyntaxKind.JSDocTypeLiteral) {
+                        break;
+                    }
+                    // falls through
                 case SyntaxKind.JSDocPropertyTag:
-                    return declareSymbolAndAddToSymbolTable(node as JSDocPropertyTag,
-                        (node as JSDocPropertyTag).isBracketed || ((node as JSDocPropertyTag).typeExpression && (node as JSDocPropertyTag).typeExpression.type.kind === SyntaxKind.JSDocOptionalType) ?
+                    return declareSymbolAndAddToSymbolTable(node as JSDocPropertyLikeTag,
+                        (node as JSDocPropertyLikeTag).isBracketed || ((node as JSDocPropertyLikeTag).typeExpression && (node as JSDocPropertyLikeTag).typeExpression.type.kind === SyntaxKind.JSDocOptionalType) ?

```

Comment:
This expression is too large.
```ts
const prop = node as JSDocPropertyLikeTag;
const flags = prop.isBracketed || prop.typeExpression && prop.typeExpression.type.kind === SyntaxKind.JSDocOptionalType
    ? SymbolFlags.Property | SymbolFlags.Optional
    : SymbolFlags.Property;
return declareSymbolAndAddToSymbolTable(prop, flags, SymbolFlags.PropertyExcludes);
```

Also, `typeExpression` is declared non-optional, but here we test for its presence.